### PR TITLE
Maps

### PR DESCRIPTION
**Pull request by AxtheDragon**
Added a map activity and show markers for each stop of the tour
This solve the issue #1

### CODE REVIEW

**Comment by Azure7111** [comment]
We got some duplicated commits and then probably also some that we don't want to keep. We should try to squash some of them.

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            si.content = i + StopInfo.CONTENT_DUMMY;
+
+            result.add(si);
+        }
+        return result;
+    }
+
+
+    /**
+     * Manipulates the map once available.
```
Comment (AxtheDragon) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: This long comment ist propably unnecessary

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        for (int i = 0; i < stoplist.size(); i++) {
+            double lat = stoplist.get(i).location.getLatitude();
+            double lng = stoplist.get(i).location.getLongitude();
+            LatLng stop = new LatLng(lat, lng);
+            Marker marker =  mMap.addMarker(new MarkerOptions()
+                    .position(stop)
+                    .title(stoplist.get(i).title)
+                    .snippet(stoplist.get(i).description)
+            );
+            StopInfo siTag = stoplist.get(i);
```
Comment (AxtheDragon) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: Could these three lines be compacted?

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.List;
+import java.util.Random;
+
+public class TourMapActivity extends FragmentActivity implements OnMapReadyCallback, GoogleMap.OnInfoWindowClickListener {
+
+    private GoogleMap mMap;
+    private String sLocationName;
+    private float randMin = -10;
+    private float randMax = +10;
+    private Random rand = new Random();
```
Comment (Azure7111) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: We can localize these three variables as they are only used in `createList`

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_tour_map);
+        stoplist = createList(10);
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+
```
Comment (Azure7111) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: We don't need this whitespace

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    private List<StopInfo> createList(int size) {
+
```
Comment (Azure7111) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: We don't need this whitespace

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java (hunk 2 above)
Comment (Azure7111) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: We are calling `stoplist.get(i)` quite often, maybe we should put it into a variable and maybe call it ``currentStop`

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                    .title(stoplist.get(i).title)
+                    .snippet(stoplist.get(i).description)
+            );
+            StopInfo siTag = stoplist.get(i);
+            marker.setTag(siTag);
+            builder.include(stop);
+        }
+        mMap.setOnInfoWindowClickListener(this);
+        mMap.moveCamera(CameraUpdateFactory.newLatLngBounds(builder.build(), 50));
+
```
Comment (Azure7111) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: We don't need this whitespace

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+   @Override
+    public void onInfoWindowClick(Marker marker) {
+        Toast.makeText(this, "Info window clicked",
+                Toast.LENGTH_SHORT).show();
```
Comment (Azure7111) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: I think we agreed to remove this toast in favor of a press of the InfoWindow to lead to the corresponding stop.

File: app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+       Intent intent = new Intent(this, StopDetailActivity.class);
+       StopInfo si = (StopInfo) marker.getTag();
+       intent.putExtra("stopInfo", si);
+       startActivity(intent);
+
```
Comment (Azure7111) [inline app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java]: We don't need this whitespace

File: app/src/release/res/values/google_maps_api.xml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    If you have previously set up a key for the debug target with the debug signing certificate,
+    you will also need to set up a key for your release certificate.
+
+    Follow the directions here:
+
+    https://developers.google.com/maps/documentation/android/signup
+
+    Once you have your key (it starts with "AIza"), replace the "google_maps_key"
+    string in this file.
+    -->
```
Comment (Azure7111) [inline app/src/release/res/values/google_maps_api.xml]: I think we can safely remove this explanatory comment now. It has served us well. :)

**Pull request merged**